File: client/app/src/main/java/com/qubacy/geoqq/domain/geo/_common/model/GeoMessage.kt
package com.qubacy.geoqq.domain.geo._common.model

import com.qubacy.geoqq.data._common.model.message.DataMessage
import com.qubacy.geoqq.domain._common.model.user.User
import com.qubacy.geoqq.domain._common.model.user.toUser

data class GeoMessage(
    val id: Long,
    val user: User,
    val text: String,
    val time: Long
) {

}

fun DataMessage.toGeoMessage(): GeoMessage {
    return GeoMessage(id, user.toUser(), text, time)
}

File: server/own_services/common/pkg/messaging/geoqq/dto/payload/geoMessage.go
package payload

import (
	domain "common/pkg/domain/geoqq"
	"time"
)

type GeoMessage struct {
	Id     float64 `json:"id"`
	Text   string  `json:"text"`
	Time   float64 `json:"time"`
	UserId float64 `json:"user-id"`

	Latitude  float64 `json:"latitude"`
	Longitude float64 `json:"longitude"`
}

func (gm *GeoMessage) ToDomain() *domain.GeoMessage {

	// some data is not translated into...

	return &domain.GeoMessage{
		Id:     uint64(gm.Id),
		Text:   gm.Text,
		Time:   time.Unix(int64(gm.Time), 0),
		UserId: uint64(gm.UserId),
	}
}
